llvm/lib/CheerpUtils/TypeOptimizer.cpp

```diff
@@ -185,6 +185,8 @@ void TypeOptimizer::gatherAllTypesInfo(const Module& M)
 				}
 				else if(const BitCastInst* BC=dyn_cast<BitCastInst>(&I))
 				{
+					if (BC->getSrcTy()->isVectorTy() || BC->getDestTy()->isVectorTy())
```

Comment:
this should be more generic and just skip anything that is not a pointer. It will solve the issue with bitcast between float and int too probably?

Reply:
This on its own did not fix the bitcast issue, but maybe all these changes together will